Mathlib/Combinatorics/SimpleGraph/Connectivity/WalkCounting.lean

```diff
@@ -259,6 +261,39 @@ lemma odd_card_iff_odd_components [Finite V] : Odd (Nat.card V) ↔
   rw [Nat.card_eq_fintype_card, Fintype.card_ofFinset]
   exact (Finset.odd_sum_iff_odd_card_odd (fun x : G.ConnectedComponent ↦ Nat.card x.supp))
 
+lemma odd_components_card_mono [Fintype V] [DecidableEq V] {G' : SimpleGraph V}
+    [DecidableRel G.Adj] (h : G ≤ G') : ({c : ConnectedComponent G' | Odd (Nat.card c.supp)}).ncard
+    ≤ ({c : ConnectedComponent G | Odd (Nat.card c.supp)}).ncard := by
+  have aux (c : G'.ConnectedComponent) (hc : Odd (Nat.card c.supp)) :=
+    Set.nonempty_of_ncard_ne_zero (by
+      intro h'
+      have := (c.odd_card_supp_iff_odd_subcomponents _ h).mp hc
+      rw [h'] at this
+      contradiction
+      : Nat.card {c' : G.ConnectedComponent | c'.supp ⊆ c.supp ∧ Odd (Nat.card c'.supp)} ≠ 0)
+  let f : {c : ConnectedComponent G' | Odd (Nat.card c.supp)} →
+      {c : ConnectedComponent G | Odd (Nat.card c.supp)} :=
+    fun ⟨c, hc⟩ ↦ ⟨(aux c hc).choose, (aux c hc).choose_spec.2⟩
+  exact Finite.card_le_of_injective f (by
+    intro c c' fcc'
+    simp only [Subtype.mk.injEq, f] at fcc'
+    exact Subtype.val_injective (ConnectedComponent.eq_of_common_vertex
+      ((fcc' ▸ (aux c.1 c.2).choose_spec.1) (ConnectedComponent.nonempty_supp _).some_mem)
+      ((aux c'.1 c'.2).choose_spec.1 (ConnectedComponent.nonempty_supp _).some_mem)))
+
+lemma odd_components_card_deleteVerts_mono [Fintype V] [DecidableEq V] (G G' : SimpleGraph V)
+    [DecidableRel G.Adj] (h : G ≤ G') (u : Set V) : Nat.card
+    ({c : ConnectedComponent ((⊤ : Subgraph G').deleteVerts u).coe | Odd (Nat.card c.supp)}) ≤
+    Nat.card ({c : ConnectedComponent ((⊤ : Subgraph G).deleteVerts u).coe | Odd (Nat.card c.supp)})
+    := by
```

Comment:
Similarly here, can you avoid starting the statement on the previous line if you are gonna need a linebreak?

Reply:
The terms here are quite long by themselves, I tried to emulate the style you used with the other one, but I'm not 100% it's correct.